src/Service/Nph/NphOrderService.php

```diff
@@ -597,6 +597,32 @@ public function saveSamplesModification(array $formData, string $type, NphOrder
         return $status;
     }
 
+    public function updateSampleModificationBulk(array $formData, string $type): bool
+    {
+        $status = true;
+        foreach ($formData as $sampleId => $checked) {
+            $sample = $this->em->getRepository(NphSample::class)->findOneBy(['sampleId' => $sampleId]);
```

Comment:
Could you run this only when the sample is checked? This will prevent unnecessary db queries.